### PR TITLE
Untangle static api dependencies

### PR DESCRIPTION
**Pull request by seank-img**
The static API header files were not actually usable due to the fact they included the Lwm2m core headers, which are not "installed" by Cmake. I have rearranged things so that the core now includes and moved all the required object files to libawa_static.a

### CODE REVIEW

File: core/src/bootstrap/lwm2m_bootstrap.c [hunk 1]
```diff
@@ -153,7 +153,7 @@ static int Lwm2mBootstrap_EndpointHandler(int type, void * ctxt, AddressType * a
 
 static void Lwm2mBootstrap_TransactionCallback(void * context, AddressType * addr, const char * responsePath, int responseCode, ContentType contentType, char * payload, int payloadLen)
 {
-    if (responseCode >= Lwm2mResult_Success && responseCode <= Lwm2mResult_SuccessContent)
+    if (responseCode >= AwaLwm2mResult_Success && responseCode <= AwaLwm2mResult_SuccessContent)
```
Comment (david-antliff-imgtec) [inline core/src/bootstrap/lwm2m_bootstrap.c]: Missing braces (style)

File: core/src/client/lwm2m_static.c [hunk 2]
```diff
@@ -164,7 +164,7 @@ AwaError AwaStaticClient_SetFactoryBootstrapInformation(AwaStaticClient * client
     {
         if (!client->Running)
         {
-            Lwm2mCore_SetFactoryBootstrap(client->Context, factoryBootstrapInformation);
+            Lwm2mCore_SetFactoryBootstrap(client->Context, (BootstrapInfo*)factoryBootstrapInformation);
```
Comment (david-antliff-imgtec) [inline core/src/client/lwm2m_static.c]: Is this cast correct?

File: core/src/common/lwm2m_observers.c [hunk 3]
```diff
@@ -115,7 +115,9 @@ void Lwm2m_MarkObserversChanged(void * ctxt, ObjectIDType objectID, ObjectInstan
             {
                 switch (definition->Type)
                 {
-                    case ResourceTypeEnum_TypeInteger: case ResourceTypeEnum_TypeFloat: case ResourceTypeEnum_TypeTime:
+                    case AwaStaticResourceType_Integer:
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_observers.c]: A "fall through" comment would help

File: core/src/common/lwm2m_result.c [hunk 4]
```diff
@@ -25,34 +25,34 @@
...
 
-int Lwm2mResult_ToAwaError(Lwm2mResult result, int defaultError)
+int AwaLwm2mResult_ToAwaError(AwaLwm2mResult result, int defaultError)
 {
     switch (result)
     {
-    case Lwm2mResult_Success: case Lwm2mResult_SuccessChanged:
-    case Lwm2mResult_SuccessContent: case Lwm2mResult_SuccessCreated:
-    case Lwm2mResult_SuccessDeleted:
+    case AwaLwm2mResult_Success: case AwaLwm2mResult_SuccessChanged:
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_result.c]: Perhaps these should be on single lines, each (style)

File: core/src/common/lwm2m_result.h [hunk 5]
```diff
@@ -30,46 +30,21 @@ extern "C" {
 
 #include <stdbool.h>
 
+#include "../../../api/include/awa/static.h"
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_result.h]: Yuck! :)

File: core/src/common/lwm2m_tlv.c [hunk 6]
```diff
@@ -932,18 +933,18 @@ static int TlvDeserialiseResourceInstance(Lwm2mTreeNode ** dest, const Definitio
...
     Lwm2mTreeNode_SetType(*dest, Lwm2mTreeNodeType_ResourceInstance);
 
-    ResourceTypeEnum resourceType = Definition_GetResourceType(registry, objectID, resourceID);
+    AwaStaticResourceType resourceType = Definition_GetResourceType(registry, objectID, resourceID);
     switch (resourceType)
     {
-        case ResourceTypeEnum_TypeInteger:
-        case ResourceTypeEnum_TypeTime:
-        case ResourceTypeEnum_TypeBoolean:
+        case AwaStaticResourceType_Integer:
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_tlv.c]: Consider a fall-through or no-break comment

File: core/src/common/lwm2m_types.h [hunk 7]
```diff
@@ -24,6 +24,7 @@
 #ifndef LWM2M_TYPES_H
 #define LWM2M_TYPES_H
 
+#include "../../../api/include/awa/static.h"
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_types.h]: Can we do anything about this, such as a better include path via cmake?

File: core/src/common/lwm2m_types.h [hunk 8]
```diff
@@ -75,29 +62,11 @@ typedef enum
...
-    uint16_t ObjectID;
-    uint16_t ObjectInstanceID;
-} ObjectLink;
-
 typedef int ObjectIDType;
 typedef int ObjectInstanceIDType;
 typedef int ResourceIDType;
 typedef int ResourceInstanceIDType;
-typedef int ResourceTypeType;
+//typedef int AwaStaticResourceType;
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_types.h]: @seank-img had a concern over why this has been removed - comments?

File: core/src/server/lwm2m_server_xml_handlers.c [hunk 9]
```diff
@@ -915,8 +915,8 @@ static void xmlif_HandleResponse(IpcCoapRequestContext * requestContext, const c
         xmlif_BackupPartiallyBuiltResponseTree(requestContext->ResponseContentNode, requestContext->ResponseObjectsTree, &backupResponseContentNode, &backupResponseObjectsTree);
     }
 
-    int responseCode = Lwm2mResult_Success;
-    if (requestContext->Result == Lwm2mResult_Success || requestContext->Result == Lwm2mResult_SuccessDeleted)
+    int responseCode = AwaLwm2mResult_Success;
+    if (requestContext->Result == AwaLwm2mResult_Success || requestContext->Result == AwaLwm2mResult_SuccessDeleted)
```
Comment (david-antliff-imgtec) [inline core/src/server/lwm2m_server_xml_handlers.c]: Parenthesis

**Comment by david-antliff-imgtec** [comment]
Reviewed, +1 contingent on comments being addressed (or at least considered)

File: core/src/common/lwm2m_types.h (hunk 7 above)
Comment (chris-dewbery-imgtec) [inline core/src/common/lwm2m_types.h]: I was just trying to solve that when I ran into the other issue. ;)

**Comment by chris-dewbery-imgtec** [comment]
Unfortunately I don't have access to sean's static-api branch, so I will merge this pull request, then create another one for the remaining changes.

**Pull request merged**